source/org/zfin/mapping/repository/HibernateLinkageRepository.java

```diff
@@ -234,87 +241,93 @@ public List<FeatureGenomeLocation> getFeatureLocations(Marker marker) {
      * @return
      */
     @Override
     public List<GenomeLocation> getGenericGenomeLocation(ZdbID markerOrFeature) {
         List<GenomeLocation> results = new ArrayList<>();
         if (markerOrFeature instanceof Marker) {
-            getGenomeLocation((Marker)markerOrFeature).stream().forEach(location -> results.add(location));
+            getGenomeLocation((Marker) markerOrFeature).stream().forEach(location -> results.add(location));
         } else if (markerOrFeature instanceof Feature) {
-            getGenomeLocation((Feature)markerOrFeature).stream().forEach(location -> results.add(location));
+            getGenomeLocation((Feature) markerOrFeature).stream().forEach(location -> results.add(location));
         }
         return results;
     }
 
     @Override
     public List<LinkageMember> getLinkageMemberForMarker(Marker marker) {
         Query query = HibernateUtil.currentSession().createQuery(
-                "from LinkageMember as linkageMember " +
-                        "where linkageMember.markerOneZdbId = :ID " +
-                        "order by linkageMember.linkage.publication.shortAuthorList");
+            "from LinkageMember as linkageMember " +
+            "where linkageMember.markerOneZdbId = :ID " +
+            "order by linkageMember.linkage.publication.shortAuthorList");
         query.setParameter("ID", marker.getZdbID());
         return (List<LinkageMember>) query.list();
     }
 
     @Override
     public List<MarkerGenomeLocation> getGenomeLocation(Marker marker) {
         Query query = HibernateUtil.currentSession().createQuery(
-                "from MarkerGenomeLocation where marker = :marker ");
+            "from MarkerGenomeLocation where marker = :marker ");
         query.setParameter("marker", marker);
         return (List<MarkerGenomeLocation>) query.list();
     }
 
     @Override
     public List<MarkerGenomeLocation> getGenomeLocationWithCoordinates(Marker marker) {
         Query query = HibernateUtil.currentSession().createQuery(
-                "from MarkerGenomeLocation mgl where marker = :marker " +
-                        " and mgl.start is not null and mgl.end is not null");
+            "from MarkerGenomeLocation mgl where marker = :marker " +
+            " and mgl.start is not null and mgl.end is not null");
         query.setParameter("marker", marker);
         return (List<MarkerGenomeLocation>) query.list();
     }
 
     @Override
     public List<MarkerGenomeLocation> getGenomeLocation(Marker marker, GenomeLocation.Source... sources) {
-        Criteria query = HibernateUtil.currentSession().createCriteria(MarkerGenomeLocation.class);
-        query.add(Restrictions.eq("marker", marker));
-        query.add(Restrictions.in("source", (Object[]) sources));
-        query.addOrder(Order.asc("chromosome"));
+        String hql = """
+            from MarkerGenomeLocation
+            where marker = :marker
+            AND source = :source
```

Comment:
should this be `source in :source` instead of `source = :source`?

Reply:
Good catch. Fixed.